adserver/analyzer/tasks.py

```diff
@@ -0,0 +1,82 @@
+"""Tasks for analyzing URLs for topics and keywords."""
+import datetime
+import logging
+
+from django.utils import timezone
+
+from ..models import Offer
+from ..models import Publisher
+from .models import AnalyzedUrl
+from .utils import get_url_analyzer_backend
+from .utils import normalize_url
+from config.celery_app import app
+
+
+log = logging.getLogger(__name__)  # noqa
+
+
+@app.task
+def analyze_url(url, publisher_slug):
+    """
+    Analyze a given URL on a publisher's site.
+
+    - Creates or updates an entry in AnalyzedUrls table
+    - Discovers keywords and topics for a URL
+    """
+    normalized_url = normalize_url(url)
+
+    publisher = Publisher.objects.filter(slug=publisher_slug).first()
+    if not publisher:
+        log.warning("Analyzing URL for nonexistent publisher: %s", publisher_slug)
+        return
+
+    backend = get_url_analyzer_backend()(url)
+    keywords = backend.analyze()  # Can be None
```

Comment:
I wonder if it should be `[]` instead when no keywords are found? Do `None` and `[]` mean different things in this case?

Reply:
Yes, they mean different things. `[]` means no keywords. `None` is a failure of some sort.